### PR TITLE
install newer sqlite version

### LINKED ISSUE #16: GDAL3.0 performances issue!
Starting in #10, we found a huge performance downgrade when doing reprojection (Warp) of dataset. 
As pointed in https://github.com/RemotePixel/amazonlinux/issues/10#issuecomment-558815307 and https://github.com/RemotePixel/amazonlinux/issues/10#issuecomment-558818578 this only happens in the containers we are building (the performance in gdal official images are fine).

## To Do
- [ ] Check PROJ compilation
- [ ] Check GDAL compilation

### PR DESCRIPTION
**Pull request by vincentsarago**
closes #16 

as mentioned in https://github.com/OSGeo/PROJ/pull/1721 PROJ 6 needs SQLite > 3.11 to have good performances!

### CODE REVIEW

**Comment by vincentsarago** [comment]
### This Branch 

```
$ ldd /opt/lib/libgdal.so | grep "sql"        
libsqlite3.so.0 => /opt/lib/libsqlite3.so.0 (0x00007f1bc7b9b000)

$ ldd /opt/lib/libproj.so | grep "sql"
libsqlite3.so.0 => /opt/lib/libsqlite3.so.0 (0x00007fae78409000)

$ ldd /opt/lib/libsqlite3.so
linux-vdso.so.1 =>  (0x00007ffd311b4000)
libz.so.1 => /lib64/libz.so.1 (0x00007f93760fd000)
libm.so.6 => /lib64/libm.so.6 (0x00007f9375dfb000)
libdl.so.2 => /lib64/libdl.so.2 (0x00007f9375bf7000)
libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f93759db000)
libc.so.6 => /lib64/libc.so.6 (0x00007f937560e000)
/lib64/ld-linux-x86-64.so.2 (0x00007f9376620000)

$ /opt/bin/sqlite3 --version
3.30.1 2019-10-10 20:19:45 18db032d058f1436ce3dea84081f4ee5a0f2259ad97301d43c426bc7f3df1b0b

 time echo "0 0" | cs2cs +proj=longlat +datum=WGS84 +to +init=epsg:2163
9473741.42      1181205.06 0.00
real    0m0.146s
user    0m0.130s
sys     0m0.000s

time GDAL_DISABLE_READDIR_ON_OPEN=EMPTY_DIR gdalwarp -q -te -11036283.891926888 10331840.239250703 -10958012.374962866 10410111.756214723 -t_srs EPSG:3857 -ts 256 256 -r bilinear -overwrite /vsicurl/https://s3.amazonaws.com/opendata.remotepixel.ca/bench_tiler/LC08_L1TP_040013_20191014_20191029_01_T1_B4.tif out.tif
real    0m3.969s
user    0m2.160s
sys     0m0.080s
```

### Master

```
$ sqlite3 --version
3.7.17 2013-05-20 00:56:22 118a3b35693b134d56ebd780123b7fd6f1497668

$ time echo "0 0" | cs2cs +proj=longlat +datum=WGS84 +to +init=epsg:2163
9473741.42      1181205.06 0.00
real    0m5.262s
user    0m4.990s
sys     0m0.280s

$ time GDAL_DISABLE_READDIR_ON_OPEN=EMPTY_DIR gdalwarp -q -te -11036283.891926888 10331840.239250703 -10958012.374962866 10410111.756214723 -t_srs EPSG:3857 -ts 256 256 -r bilinear -overwrite /vsicurl/https://s3.amazonaws.com/opendata.remotepixel.ca/bench_tiler/LC08_L1TP_040013_20191014_20191029_01_T1_B4.tif out.tif
real    0m3.974s
user    0m1.970s
sys     0m0.120s
```

Updating SQLite to > 3.11 (3.30) did fix performances of PROJ (cs2cs command) but has no influence on gdalwarp.

There might be something else going on :-( 

cc @rouault

**Comment by rouault** [comment]
> but has no influence on gdalwarp.

what about ldd on gdalwarp / libgdal.so to check which sqlite you're linking against ?

**Comment by vincentsarago** [comment]
> what about ldd on gdalwarp / libgdal.so to check which sqlite you're linking against ?

```
$ ldd /opt/bin/cs2cs | grep "sql"
        libsqlite3.so.0 => /opt/lib/libsqlite3.so.0 (0x00007f0cc6415000)
$ ldd /opt/lib/libgdal.so | grep "sqlite"
        libsqlite3.so.0 => /opt/lib/libsqlite3.so.0 (0x00007f16d615c000)
$ ldd /opt/bin/gdalwarp | grep "sqlite"
        libsqlite3.so.0 => /usr/lib64/libsqlite3.so.0 (0x00007fd24d448000)
```

So `gdalwarp` is linked to the old sqlite while libgdal and `cs2cs` are linked to the new one.

Right on @rouault, when I force `/opt/lib` to be first in `LD_LIBRARY_PATH` it then work fine. But still wondering why `cs2cs` found the good sqlite while gdalwarp doesn't.

```
export LD_LIBRARY_PATH=/opt/lib:$LD_LIBRARY_PATH
$ ldd /opt/bin/gdalwarp | grep "sqlite"
        libsqlite3.so.0 => /opt/lib/libsqlite3.so.0 (0x00007f86d7f84000)

$ time GDAL_DISABLE_READDIR_ON_OPEN=EMPTY_DIR gdalwarp -q -te -11036283.891926888 10331840.239250703 -10958012.374962866 10410111.756214723 -t_srs EPSG:3857 -ts 256 256 -r bilinear -overwrite /vsicurl/https://s3.amazonaws.com/opendata.remotepixel.ca/bench_tiler/LC08_L1TP_040013_20191014_20191029_01_T1_B4.tif out.tif
real    0m0.780s
user    0m0.120s
sys     0m0.040s
```

**Comment by vincentsarago** [comment]
Ok, seems that PROJ user `rpath` by default to its binaries while gdal doesn't. I think I know what to do next. thanks a lot @rouault 

```
$ readelf -d /opt/bin/cs2cs | grep "rpath"
 0x000000000000000f (RPATH)              Library rpath: [/opt/lib]

$ readelf -d /opt/bin/gdalwarp | grep "rpath"
empty

$ readelf -d /opt/lib/libgdal.so | grep "rpath"
 0x000000000000000f (RPATH)              Library rpath: [$ORIGIN]
```

```

File: base/gdal3.0/Dockerfile [hunk 1]
```diff
@@ -161,19 +152,16 @@ RUN mkdir /tmp/patchelf \
...
-# # Move /lib64 (libjpeg) to /lib
+# Move /lib64 (libjpeg) to /lib
 RUN mv $PREFIX/lib64/lib* $PREFIX/lib/
 
-# Set $ORIGIN
-RUN cd $PREFIX/lib/ \
-  && patchelf --remove-rpath libgdal.so \
-  && patchelf --force-rpath --set-rpath '$ORIGIN' ./libgdal.so
+# FIX 
+RUN for i in $PREFIX/bin/*; do  patchelf --force-rpath --set-rpath '$ORIGIN/../lib' $i; done
```
Comment (vincentsarago) [inline base/gdal3.0/Dockerfile]: Here I'm making sure the binaries are fetching libs from `$ORIGIN/../lib` first

**Pull request merged**